// Copyright (c) 2021 Huawei Device Co., Ltd.
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";
option optimize_for = LITE_RUNTIME;

message HiperfPluginConfig {
    bool is_root = 1;
    string outfile_name = 2; // set output file name, must be set for merging to one file
    string record_args = 3; // set arguments of hiperf record
    enum LogLevel {
        NO_LOG = 0;
        MUCH = 1;
        VERBOSE = 2;
        DEBUG = 3;
    };
    LogLevel log_level = 4; // set log level, default is NO_LOG
}